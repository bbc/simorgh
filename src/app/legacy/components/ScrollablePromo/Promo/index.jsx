import React, { useContext } from 'react';
import { arrayOf, shape, string, oneOfType, object, func } from 'prop-types';
import styled from '@emotion/styled';
import path from 'ramda/src/path';
import pathOr from 'ramda/src/pathOr';
import { getPica } from '#psammead/gel-foundations/src/typography';
import { getSerifBold } from '#psammead/psammead-styles/src/font-styles';
import { Link } from '#psammead/psammead-story-promo/src';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
} from '#psammead/gel-foundations/src/spacings';
import {
  GEL_GROUP_0_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';
import filterForBlockType from '#lib/utilities/blockHandlers';
import useOperaMiniDetection from '#hooks/useOperaMiniDetection';
import PromoTimestamp from '#components/Promo/timestamp';
import { ServiceContext } from '../../../../contexts/ServiceContext';

const StyledLink = styled(Link)`
  ${({ script }) => script && getPica(script)}
  ${({ service }) => service && getSerifBold(service)}
  width: 100%;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  text-decoration: none;

  overflow-x: hidden;
  overflow-y: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  color: ${({ theme }) =>
    theme.isDarkUi ? theme.palette.GREY_10 : theme.palette.GREY_8};
  &:visited {
    color: ${props => props.theme.palette.GREY_6};
  }
`;

const PromoBox = styled.div`
  position: relative;
  background-color: ${({ theme }) =>
    theme.isDarkUi ? theme.palette.GREY_3 : theme.palette.WHITE};
  padding: ${GEL_SPACING_DBL};
  margin-bottom: ${GEL_SPACING_TRPL};
  @media (min-width: ${GEL_GROUP_0_SCREEN_WIDTH_MIN}) {
    width: 14.8125rem;
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    width: 11.125rem;
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    width: 12.6875rem;
  }
`;

const OperaPromoBox = styled.div`
  position: relative;
  background-color: ${({ theme }) =>
    theme.isDarkUi ? theme.palette.GREY_3 : theme.palette.WHITE};
  padding: ${GEL_SPACING_DBL};
  margin-bottom: ${GEL_SPACING_DBL};
  width: calc(100% - ${GEL_SPACING});
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    width: calc(100% - ${GEL_SPACING_DBL});
  }
`;

const TimeStamp = styled(PromoTimestamp)`
  margin-top: ${GEL_SPACING};
  color: ${({ theme }) => theme.isDarkUi && theme.palette.GREY_6};
`;

const Promo = ({ block, onClick }) => {
  const { script, service, serviceDatetimeLocale } = useContext(ServiceContext);
  const textBlock = filterForBlockType(
    pathOr({}, ['model', 'blocks'], block),
    'text',
  );
  const aresLinkBlock = filterForBlockType(
    pathOr({}, ['model', 'blocks'], block),
    'aresLink',
  );
  const href = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'locator'],
    textBlock,
  );
  const title = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'text'],
    textBlock,
  );
  const timestamp = path(
    ['model', 'blocks', '0', 'model', 'timestamp'],
    aresLinkBlock,
  );

  const isOperaMini = useOperaMiniDetection();

  const WrapperPromoBox = isOperaMini ? OperaPromoBox : PromoBox;

  return (
    <WrapperPromoBox>
      <StyledLink
        href={href}
        service={service}
        script={script}
        onClick={onClick}
      >
        {title}
      </StyledLink>
      {timestamp && (
        <TimeStamp serviceDatetimeLocale={serviceDatetimeLocale}>
          {timestamp}
        </TimeStamp>
      )}
    </WrapperPromoBox>
  );
};

Promo.propTypes = {
  block: shape({
    model: shape({
      blocks: arrayOf(oneOfType([string, object])),
    }).isRequired,
  }).isRequired,
  onClick: func.isRequired,
};

export default Promo;
