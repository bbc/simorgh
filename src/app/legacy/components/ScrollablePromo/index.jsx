import React, { useContext } from 'react';
import { arrayOf, shape, string, oneOfType, object, number } from 'prop-types';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';

import { getDoublePica } from '#psammead/gel-foundations/src/typography';
import { getSansRegular } from '#psammead/psammead-styles/src/font-styles';
import styled from '@emotion/styled';
import path from 'ramda/src/path';
import pathOr from 'ramda/src/pathOr';
import isEmpty from 'ramda/src/isEmpty';
import tail from 'ramda/src/tail';
import {
  GEL_GROUP_0_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';
import { GridItemMediumNoMargin } from '#components/Grid';
import useViewTracker from '#hooks/useViewTracker';
import useClickTrackerHandler from '#hooks/useClickTrackerHandler';
import idSanitiser from '#lib/utilities/idSanitiser';
import { ServiceContext } from '../../../contexts/ServiceContext';
import Promo from './Promo';
import PromoList from './PromoList';

const PromoWrapper = styled.div`
  ${({ dir }) => `margin-${dir === 'ltr' ? 'left' : 'right'}: ${GEL_SPACING};`}
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    ${({ dir }) =>
      `margin-${dir === 'ltr' ? 'left' : 'right'}: ${GEL_SPACING_DBL};`}
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    ${({ dir }) => `margin-${dir === 'ltr' ? 'left' : 'right'}: 0;`}
  }
`;

const LabelComponent = styled.strong`
  display: block;
  ${({ script }) => script && getDoublePica(script)};
  ${({ service }) => getSansRegular(service)}
  margin-bottom: ${GEL_SPACING_DBL};
  color: ${props => props.theme.palette.SHADOW};

  ${({ dir }) =>
    `
    @media (min-width: ${GEL_GROUP_0_SCREEN_WIDTH_MIN}){
      margin-${dir === 'ltr' ? 'left' : 'right'}: ${GEL_SPACING};
    }
    @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}){
      margin-${dir === 'ltr' ? `left` : `right`}: ${GEL_SPACING_DBL};  
    }
    @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}){
        margin-${dir === 'ltr' ? `left` : `right`}: 0;
    }
`}
`;

const ScrollablePromo = ({ blocks, blockGroupIndex }) => {
  const { script, service, dir, translations } = useContext(ServiceContext);

  const eventTrackingData = {
    componentName: `edoj${blockGroupIndex}`,
    format: 'CHD=edoj',
  };

  const viewRef = useViewTracker(eventTrackingData);
  const handleClickTracking = useClickTrackerHandler(eventTrackingData);

  if (!blocks || isEmpty(blocks)) {
    return null;
  }

  const title =
    blocks[0].type === 'title' &&
    path(
      ['0', 'model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'text'],
      blocks,
    );

  const blocksWithoutTitle = blocks[0].type === 'title' ? tail(blocks) : blocks;

  const isSingleItem = blocksWithoutTitle.length === 1;

  const ariaLabel = title && idSanitiser(title);

  const a11yAttributes = {
    as: 'section',
    role: 'region',
    ...(ariaLabel
      ? { 'aria-labelledby': ariaLabel }
      : {
          'aria-label': pathOr(
            'Related Content',
            ['relatedContent'],
            translations,
          ),
        }),
  };

  return (
    <GridItemMediumNoMargin {...a11yAttributes}>
      {title && (
        <LabelComponent
          id={ariaLabel}
          data-testid="eoj-recommendations-heading"
          script={script}
          service={service}
          dir={dir}
        >
          {title}
        </LabelComponent>
      )}
      {isSingleItem ? (
        <PromoWrapper dir={dir} ref={viewRef}>
          <Promo block={blocksWithoutTitle[0]} onClick={handleClickTracking} />
        </PromoWrapper>
      ) : (
        <PromoList
          blocks={blocksWithoutTitle}
          viewTracker={viewRef}
          onClick={handleClickTracking}
        />
      )}
    </GridItemMediumNoMargin>
  );
};

ScrollablePromo.propTypes = {
  blocks: arrayOf(
    shape({
      type: string.isRequired,
      model: shape({
        blocks: arrayOf(oneOfType([string, object])),
      }).isRequired,
    }),
  ).isRequired,
  blockGroupIndex: number,
};

ScrollablePromo.defaultProps = {
  blockGroupIndex: null,
};

export default ScrollablePromo;
