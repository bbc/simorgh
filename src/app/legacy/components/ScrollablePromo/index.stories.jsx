import React from 'react';
import { withServicesKnob } from '#psammead/psammead-storybook-helpers/src';
import { withKnobs } from '@storybook/addon-knobs';
import styled from '@emotion/styled';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { ServiceContextProvider } from '../../../contexts/ServiceContext';
import ScrollablePromo from '.';
import {
  threeLinks,
  oneLinkOnly,
  oneLinkWithNoTitle,
  oneLinkWithTimestamp,
  moreThanThreeLinks,
  twoLinksWithNoImages,
  truncatedTextInSingleLink,
  arabicText,
} from './helpers/fixtureData';
import ThemeProvider from '../../../components/ThemeProvider';

const BackGround = styled.div`
  background-color: #f6f6f6;
  padding: 2rem;
`;
// eslint-disable-next-line react/prop-types
const ScrollablePromoComponent = ({ data, service, script, dir }) => (
  <ThemeProvider service={service}>
    <ToggleContextProvider>
      <BackGround>
        <ServiceContextProvider service={service} script={script} dir={dir}>
          <ScrollablePromo blocks={data} />
        </ServiceContextProvider>
      </BackGround>
    </ToggleContextProvider>
  </ThemeProvider>
);

export default {
  title: 'components/scrollable promo component',
  ScrollablePromoComponent,
  decorators: [withKnobs, withServicesKnob()],
};

export const ThreeLinks = props => (
  <ScrollablePromoComponent data={threeLinks} {...props} />
);

export const OnlyOneLink = props => (
  <ScrollablePromoComponent data={oneLinkOnly} {...props} />
);

export const OneLinkWithNoTitle = props => (
  <ScrollablePromoComponent data={oneLinkWithNoTitle} {...props} />
);

export const MoreThanThreeLinks = props => (
  <ScrollablePromoComponent data={moreThanThreeLinks} {...props} />
);

export const NoImagesInData = props => (
  <ScrollablePromoComponent data={twoLinksWithNoImages} {...props} />
);

export const TruncatedTextInSingleLink = props => (
  <ScrollablePromoComponent data={truncatedTextInSingleLink} {...props} />
);

export const ArabicText = props => (
  <ScrollablePromoComponent data={arabicText} {...props} />
);

export const WithTimestamp = props => (
  <ScrollablePromoComponent data={oneLinkWithTimestamp} {...props} />
);
