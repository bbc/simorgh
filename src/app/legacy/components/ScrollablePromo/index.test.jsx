import React from 'react';
import * as viewTracking from '#hooks/useViewTracker';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import * as clickTracking from '#hooks/useClickTrackerHandler';
import { render } from '../../../components/react-testing-library-with-providers';
import { ServiceContext } from '../../../contexts/ServiceContext';
import {
  threeLinks,
  oneLinkOnly,
  oneLinkWithNoTitle,
  moreThanThreeLinks,
} from './helpers/fixtureData';
import ScrollablePromo from '.';
import { edOjA, edOjB } from './fixtures';

// eslint-disable-next-line react/prop-types
const ScrollablePromoWithContext = ({ blocks, blockGroupIndex }) => (
  <ToggleContextProvider>
    <ServiceContext.Provider value={{ service: 'news' }}>
      <ScrollablePromo blocks={blocks} blockGroupIndex={blockGroupIndex} />
    </ServiceContext.Provider>
  </ToggleContextProvider>
);

describe('ScrollablePromo', () => {
  it('should return null if no data is passed', () => {
    const { container } = render(<ScrollablePromoWithContext blocks={{}} />);
    expect(container).toBeEmptyDOMElement();
  });

  it('should render max 3 promo items', () => {
    const { getAllByRole } = render(
      <ScrollablePromoWithContext blocks={moreThanThreeLinks} />,
    );
    expect(getAllByRole('listitem').length).toEqual(3);
  });

  it('should render single promo item', () => {
    const { container } = render(
      <ScrollablePromoWithContext blocks={oneLinkOnly} />,
    );
    expect(container.childElementCount).toEqual(1);
  });

  it('should render single promo item with a title', () => {
    const { container, getByTestId } = render(
      <ScrollablePromoWithContext blocks={oneLinkOnly} />,
    );
    expect(container.childElementCount).toEqual(1);
    expect(getByTestId('eoj-recommendations-heading')).toBeInTheDocument();
  });

  it('should render single promo item without a title', () => {
    const { container, queryByTestId } = render(
      <ScrollablePromoWithContext blocks={oneLinkWithNoTitle} />,
    );
    expect(container.childElementCount).toEqual(1);
    expect(
      queryByTestId('eoj-recommendations-heading'),
    ).not.toBeInTheDocument();
  });

  it('should not render a list when there is only one promo', () => {
    const { queryByRole } = render(
      <ScrollablePromoWithContext blocks={oneLinkOnly} />,
    );

    expect(queryByRole('list')).not.toBeInTheDocument();
    expect(queryByRole('listitem')).not.toBeInTheDocument();
  });

  it('should render unordered list if more than 1 item', () => {
    const { queryByRole, getAllByRole } = render(
      <ScrollablePromoWithContext blocks={threeLinks} />,
    );
    expect(queryByRole('list')).toBeInTheDocument();
    expect(getAllByRole('listitem').length).toEqual(3);
  });

  describe('event tracking in editorial onward journeys', () => {
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should call the view tracking hook with the correct params with one editorial onward journey', () => {
      const viewTrackerSpy = jest.spyOn(viewTracking, 'default');

      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
        />,
      );

      expect(viewTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
    });

    it('should call the view tracking hook with the correct params with multiple editorial onward journeys', () => {
      const viewTrackerSpy = jest.spyOn(viewTracking, 'default');
      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
        />,
      );
      render(
        <ScrollablePromoWithContext
          blocks={edOjB.model.blocks}
          blockGroupIndex={2}
        />,
      );

      expect(viewTrackerSpy).toHaveBeenCalledTimes(2);
      expect(viewTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
      expect(viewTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj2',
        format: 'CHD=edoj',
      });
    });

    it('should call the click tracking hook with one editorial onward journey', () => {
      const clickTrackerSpy = jest.spyOn(clickTracking, 'default');
      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
        />,
      );

      expect(clickTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
    });

    it('should call the click tracking hook with multiple editorial onward journeys', () => {
      const clickTrackerSpy = jest.spyOn(clickTracking, 'default');
      render(
        <ScrollablePromoWithContext
          blocks={edOjA.model.blocks}
          blockGroupIndex={1}
        />,
      );
      render(
        <ScrollablePromoWithContext
          blocks={edOjB.model.blocks}
          blockGroupIndex={2}
        />,
      );

      expect(clickTrackerSpy).toHaveBeenCalledTimes(2);
      expect(clickTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj1',
        format: 'CHD=edoj',
      });
      expect(clickTrackerSpy).toHaveBeenCalledWith({
        componentName: 'edoj2',
        format: 'CHD=edoj',
      });
    });
  });

  it('it should match a11y snapshot for single card', () => {
    const { container } = render(
      <ScrollablePromoWithContext blocks={oneLinkOnly} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('it should match a11y snapshot for list', () => {
    const { container } = render(
      <ScrollablePromoWithContext blocks={threeLinks} />,
    );
    expect(container).toMatchSnapshot();
  });
  it('it should match a11y snapshot for list with no title', () => {
    const { container } = render(
      <ScrollablePromoWithContext blocks={oneLinkWithNoTitle} />,
    );
    expect(container).toMatchSnapshot();
  });
});
