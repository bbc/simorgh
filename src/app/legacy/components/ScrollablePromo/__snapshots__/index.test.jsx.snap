// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollablePromo it should match a11y snapshot for list 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-2 {
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-bottom: 1rem;
  color: #3F3F42;
}

@media (min-width: 0rem) {
  .emotion-2 {
    margin-right: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-2 {
    margin-right: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-2 {
    margin-right: 0;
  }
}

.emotion-4 {
  list-style: none;
  padding-right: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: scroll;
  scroll-behavior: auto;
  -webkit-overflow-scrolling: touch;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.emotion-4::-webkit-scrollbar {
  display: none;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media (min-width: 0rem) {
  .emotion-6 {
    margin-right: 0.5rem;
  }

  .emotion-6:first-child {
    margin-right: 0.5rem;
  }

  .emotion-6:last-child {
    margin-left: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-6 {
    margin-right: 1rem;
  }

  .emotion-6:first-child {
    margin-right: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-6 {
    margin-right: 1rem;
  }

  .emotion-6:first-child {
    margin-right: 0;
  }
}

.emotion-8 {
  position: relative;
  background-color: #FFFFFF;
  padding: 1rem;
  margin-bottom: 1.5rem;
}

@media (min-width: 0rem) {
  .emotion-8 {
    width: 14.8125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    width: 11.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-8 {
    width: 12.6875rem;
  }
}

.emotion-10 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-wrap: break-word;
  display: inline-block;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  width: 100%;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  overflow-y: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  color: #202224;
}

.emotion-10:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.emotion-10:hover,
.emotion-10:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10:visited {
  color: #6E6E73;
}

.emotion-10:hover,
.emotion-10:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10:visited {
  color: #545658;
}

<div>
  <section
    aria-labelledby="Show-all-links-no-images"
    class="emotion-0 emotion-1"
    dir="ltr"
    role="region"
  >
    <strong
      class="emotion-2 emotion-3"
      data-testid="eoj-recommendations-heading"
      id="Show-all-links-no-images"
    >
      Show all links (no images)
    </strong>
    <ul
      class="emotion-4 emotion-5"
      role="list"
    >
      <li
        class="emotion-6 emotion-7"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <a
            class="emotion-10 emotion-11"
            href="https://www.bbc.co.uk/pidgin"
          >
            Show headline
          </a>
        </div>
      </li>
      <li
        class="emotion-6 emotion-7"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <a
            class="emotion-10 emotion-11"
            href="https://www.bbc.com/mundo"
          >
            Link with Headline
          </a>
        </div>
      </li>
      <li
        class="emotion-6 emotion-7"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <a
            class="emotion-10 emotion-11"
            href="https://www.bbc.com/brasil"
          >
            Brasil homepage
          </a>
        </div>
      </li>
    </ul>
  </section>
</div>
`;

exports[`ScrollablePromo it should match a11y snapshot for list with no title 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-2 {
  margin-right: 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-2 {
    margin-right: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-2 {
    margin-right: 0;
  }
}

.emotion-4 {
  position: relative;
  background-color: #FFFFFF;
  padding: 1rem;
  margin-bottom: 1.5rem;
}

@media (min-width: 0rem) {
  .emotion-4 {
    width: 14.8125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    width: 11.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-4 {
    width: 12.6875rem;
  }
}

.emotion-6 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-wrap: break-word;
  display: inline-block;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  width: 100%;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  overflow-y: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  color: #202224;
}

.emotion-6:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.emotion-6:hover,
.emotion-6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:visited {
  color: #6E6E73;
}

.emotion-6:hover,
.emotion-6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:visited {
  color: #545658;
}

<div>
  <section
    aria-label="Related Content"
    class="emotion-0 emotion-1"
    dir="ltr"
    role="region"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        <a
          class="emotion-6 emotion-7"
          href="https://www.bbc.com/pidgin"
        >
          Single link
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`ScrollablePromo it should match a11y snapshot for single card 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 40%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(10, 1fr);
      grid-column-end: span 10;
      grid-column-start: 5;
    }
  }
}

.emotion-2 {
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  margin-bottom: 1rem;
  color: #3F3F42;
}

@media (min-width: 0rem) {
  .emotion-2 {
    margin-right: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-2 {
    margin-right: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-2 {
    margin-right: 0;
  }
}

.emotion-4 {
  margin-right: 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-4 {
    margin-right: 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-4 {
    margin-right: 0;
  }
}

.emotion-6 {
  position: relative;
  background-color: #FFFFFF;
  padding: 1rem;
  margin-bottom: 1.5rem;
}

@media (min-width: 0rem) {
  .emotion-6 {
    width: 14.8125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-6 {
    width: 11.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-6 {
    width: 12.6875rem;
  }
}

.emotion-8 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-wrap: break-word;
  display: inline-block;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  width: 100%;
  overflow-wrap: break-word;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  overflow-y: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  color: #202224;
}

.emotion-8:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.emotion-8:hover,
.emotion-8:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-8:visited {
  color: #6E6E73;
}

.emotion-8:hover,
.emotion-8:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-8:visited {
  color: #545658;
}

<div>
  <section
    aria-labelledby="Single-link"
    class="emotion-0 emotion-1"
    dir="ltr"
    role="region"
  >
    <strong
      class="emotion-2 emotion-3"
      data-testid="eoj-recommendations-heading"
      id="Single-link"
    >
      Single link
    </strong>
    <div
      class="emotion-4 emotion-5"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <a
          class="emotion-8 emotion-9"
          href="https://www.bbc.com/pidgin"
        >
          Single link
        </a>
      </div>
    </div>
  </section>
</div>
`;
